packages/kbn-discover-utils/src/data_types/logs/logs_context_service.ts

```diff
@@ -28,15 +27,36 @@ export const DEFAULT_ALLOWED_LOGS_BASE_PATTERNS = [
   'winlogbeat',
 ];
 
-export const createLogsContextService = (_deps: LogsContextServiceDeps = {}) => {
-  // This is initially an hard-coded set of well-known base patterns,
-  // we can extend this allowed list with any setting coming from uiSettings
-  const ALLOWED_LOGS_DATA_SOURCES = [createRegExpPatternFrom(DEFAULT_ALLOWED_LOGS_BASE_PATTERNS)];
+export const DEFAULT_ALLOWED_LOGS_BASE_PATTERNS_REGEXP = createRegExpPatternFrom(
+  DEFAULT_ALLOWED_LOGS_BASE_PATTERNS
+);
 
+export const createLogsContextService = async ({
+  logsDataAccessPlugin,
+}: LogsContextServiceDeps) => {
+  let logSources;
+
+  if (logsDataAccessPlugin) {
+    const logSourcesService = logsDataAccessPlugin.services.logSourcesService;
+    logSources = (await logSourcesService.getLogSources())
+      .map((logSource) => logSource.indexPattern)
+      .join(',') // TODO: Will be replaced by helper in: https://github.com/elastic/kibana/pull/192003
+      .split(',');
```

Comment:
> [!NOTE]
> Isn't the `.join(',').split(',')` reverting the string back to the same array created with the `.map` call?

Reply:
No, not quite. This is what's happening here:

![Screenshot 2024-09-12 at 14 09 27](https://github.com/user-attachments/assets/a3245e7c-8435-4f7b-b490-270c756bc847)

Could also use a `toString()`, or various other methods, for the string conversion.

Reply:
Oh I see, codewise makes sense 👌 OTOH, is it possible from the advanced settings to have log sources comma-separated? Isn't it the setting a single string, which is then separated into an array by the comma character?
Not blocking here, I'm trying to understand how we can end up in a scenario like the one in your screenshot.

Reply:
It is registered as a `schema.arrayOf(schema.string())` yep. That will be fed back to us as `['test-*', 'another-test-*']` from the UI settings client if the user has made a change via the UI. With the defaults it's `['logs-*-*,logs-*,filebeat-*,kibana_sample_data_logs*']` purely because I set the default as `['logs-*-*,logs-*,filebeat-*,kibana_sample_data_logs*']` originally. Both work with this code though (sorry my screenshot was a bit rubbish, I forgot the array 😅). What I could do, purely for alignment, is make the defaults comma separated strings.

Reply:
Thanks for clarifying this, makes sense to extra preventive in that scenario!